logging/logging.go

```diff
@@ -0,0 +1,108 @@
+package logging
+
+import (
+	"log/slog"
+	"net/http"
+	"strings"
+	"time"
+
+	azcorePolicy "github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+)
+
+var resourceTypes = map[string]bool{
+	"resourcegroups":   true,
+	"storageaccounts":  true,
+	"operationresults": true,
+	"asyncoperations":  true,
+}
+
+type LogRequestParams struct {
+	Logger    *slog.Logger
+	StartTime time.Time
+	Request   interface{}
+	Response  *http.Response
+	Error     error
+}
+
+// Shared logging function for REST API interactions
+func GetMethodInfo(method string, rawURL string) string {
+	url := strings.Split(rawURL, "?api-version")
+	parts := strings.Split(url[0], "/")
+	resource := url[0]
+	counter := 0
+	// Start from the end of the split path and move backward
+	// to get nested resource type
+	for counter = len(parts) - 1; counter >= 0; counter-- {
+		currToken := parts[counter]
+		if resourceTypes[strings.ToLower(currToken)] {
+			resource = currToken
```

Comment:
Use normalized resource type (e.g., lower case) to reduce variants.

Reply:
Yes, currently our resource map is populated with all lowercase resource name. We convert each token to lowercase when comparing to the elements in the map.

Reply:
I see, originally I wanted to preserve the name from the URL and convert to lowercase for purpose of comparing to map. Done, it is now all in lowercase.